src/core/policy.rs

```diff
@@ -194,6 +194,191 @@ impl AccessPolicy {
         Ok(access_policy)
     }
 
+    ///This function is finding the right closing parenthesis in the boolean
+    /// expression given as a string
+    fn find_next_parenthesis(boolean_expression: &str) -> Result<usize, FormatErr> {
+        let mut count = 0;
+        let mut right_closing_parenthesis = 0;
+        // Skip first parenthesis
+        // let boolean_expression=&boolean_expression[1..];
+        for (index, c) in boolean_expression.chars().enumerate() {
+            match c {
+                '(' => count += 1,
+                ')' => count -= 1,
+                _ => {}
+            };
+            if count < 0 {
+                right_closing_parenthesis = index;
+                break;
+            }
+        }
+        if right_closing_parenthesis == 0 {
+            return Err(FormatErr::InvalidBooleanExpression(format!(
+                "Missing closing parenthesis in boolean expression {boolean_expression} "
+            )));
+        }
+        Ok(right_closing_parenthesis)
+    }
+
+    fn decompose_expression(
```

Comment:
can you add a short description (with an example if possible)?

Reply:
done